src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -770,66 +846,107 @@ private BoundPattern BindVarPattern(VarPatternSyntax node, TypeSymbol inputType,
+        private BoundPattern BindVarDesignation(
+            VariableDesignationSyntax node,
+            TypeSymbol inputType,
+            uint inputValEscape,
+            bool hasErrors,
+            DiagnosticBag diagnostics)
         {
-            switch (designation.Kind())
+            switch (node.Kind())
             {
                 case SyntaxKind.DiscardDesignation:
                     {
-                        return new BoundDiscardPattern(designation, inputType);
+                        return new BoundDiscardPattern(node, inputType);
                     }
                 case SyntaxKind.SingleVariableDesignation:
                     {
                         BindPatternDesignation(
-                            node: node, designation: designation, declType: inputType, inputValEscape: inputValEscape, typeSyntax: null, diagnostics: diagnostics,
-                            hasErrors: ref hasErrors, variableSymbol: out Symbol variableSymbol, variableAccess: out BoundExpression variableAccess);
+                            designation: node, declType: inputType, inputValEscape: inputValEscape, typeSyntax: null, diagnostics: diagnostics, hasErrors: ref hasErrors,
+                            variableSymbol: out Symbol variableSymbol, variableAccess: out BoundExpression variableAccess);
                         var boundOperandType = new BoundTypeExpression(syntax: node, aliasOpt: null, type: inputType); // fake a type expression for the variable's type
                         // We continue to use a BoundDeclarationPattern for the var pattern, as they have more in common.
                         return new BoundDeclarationPattern(
-                            designation.Parent == node ? node : (SyntaxNode)designation, // for `var x` use whole pattern, otherwise use designation for the syntax
+                            node.Parent.Kind() == SyntaxKind.VarPattern ? node.Parent : node, // for `var x` use whole pattern, otherwise use designation for the syntax
                             variableSymbol, variableAccess, boundOperandType, isVar: true, inputType: inputType, hasErrors: hasErrors);
                     }
                 case SyntaxKind.ParenthesizedVariableDesignation:
                     {
-                        var tupleDesignation = (ParenthesizedVariableDesignationSyntax)designation;
+                        var tupleDesignation = (ParenthesizedVariableDesignationSyntax)node;
                         var subPatterns = ArrayBuilder<BoundSubpattern>.GetInstance(tupleDesignation.Variables.Count);
                         MethodSymbol deconstructMethod = null;
-                        if (inputType.IsTupleType)
+                        var strippedInputType = inputType.StrippedType();
+
+                        if (IsZeroElementTupleType(strippedInputType))
```

Comment:
>IsZeroElementTupleType(strippedInputType) [](start = 28, length = 41)

It looks like the only difference between this case and the `IsTupleType` case below is `elementTypes`. If so, would it make sense to extract a local function to share code where the caller passes in `elementTypes`?